pkg/network/network_dynamic.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 Nubificus LTD.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network
+
+import (
+	"fmt"
+	"os/user"
+	"strconv"
+	"strings"
+
+	"github.com/nubificus/urunc/internal/constants"
+	"github.com/vishvananda/netlink"
+)
+
+type DynamicNetwork struct {
+}
+
+// NetworkSetup checks if any tap device is available in the current netns. If not,
+// creates a new tap device and sets tc rules between veth interface
```

Comment:
Maybe a better rephrasing:
"sets TC rules between the veth interface and the tap device inside the namespace".

Reply:
Fixed it.